test-tests/shared/src/test/scala/zio/test/environment/RandomSpec.scala

```diff
@@ -55,117 +58,121 @@ object RandomSpec
-    val seed    = SRandom.nextLong()
-    val sRandom = new SRandom(seed)
-    for {
-      testRandom <- TestRandom.makeTest(DefaultData)
-      _          <- testRandom.setSeed(seed)
-      actual     <- UIO.foreach(List.fill(100)(()))(_ => f(testRandom))
-      expected   = List.fill(100)(g(sRandom))
-    } yield assert(actual, equalTo(expected))
-  }
+  def forAllEqual[A](
+    f: Test => UIO[A]
+  )(g: SRandom => A): ZIO[Random, Nothing, TestResult] =
+    checkM(Gen.anyLong) { seed =>
+      for {
+        sRandom    <- ZIO.effectTotal(new SRandom(seed))
+        testRandom <- TestRandom.makeTest(DefaultData)
+        _          <- testRandom.setSeed(seed)
+        actual     <- UIO.foreach(List.fill(100)(()))(_ => f(testRandom))
+        expected   <- ZIO.effectTotal(List.fill(100)(g(sRandom)))
+      } yield assert(actual, equalTo(expected))
+    }
 
-  def forAllEqualBytes: ZIO[Any, Nothing, TestResult] = {
-    val seed    = SRandom.nextLong()
-    val sRandom = new SRandom(seed)
-    for {
-      testRandom <- TestRandom.makeTest(DefaultData)
-      _          <- testRandom.setSeed(seed)
-      actual     <- UIO.foreach(List.range(0, 100))(testRandom.nextBytes(_))
-      expected = List.range(0, 100).map(new Array[Byte](_)).map { arr =>
-        sRandom.nextBytes(arr)
-        Chunk.fromArray(arr)
-      }
-    } yield assert(actual, equalTo(expected))
-  }
+  def forAllEqualBytes: ZIO[Random, Nothing, TestResult] =
+    checkM(Gen.anyLong) { seed =>
+      for {
+        sRandom    <- ZIO.effectTotal(new SRandom(seed))
+        testRandom <- TestRandom.makeTest(DefaultData)
+        _          <- testRandom.setSeed(seed)
+        actual     <- UIO.foreach(0 to 100)(testRandom.nextBytes(_))
+        expected <- ZIO.effectTotal((0 to 100).map(new Array[Byte](_)).map { arr =>
+                     sRandom.nextBytes(arr)
+                     Chunk.fromArray(arr)
+                   })
+      } yield assert(actual, equalTo(expected))
+    }
 
-  def forAllEqualGaussian: ZIO[Any, Nothing, TestResult] = {
-    val seed    = SRandom.nextLong()
-    val sRandom = new SRandom(seed)
-    for {
-      testRandom <- TestRandom.makeTest(DefaultData)
-      _          <- testRandom.setSeed(seed)
-      actual     <- UIO.foreach(List.fill(100)(()))(_ => testRandom.nextGaussian)
-      expected   = List.fill(100)(sRandom.nextGaussian)
-    } yield assert(actual.zip(expected).forall { case (x, y) => math.abs(x - y) < 0.01 }, equalTo(true))
-  }
+  def forAllEqualGaussian: ZIO[Random, Nothing, TestResult] =
+    checkM(Gen.anyLong) { seed =>
+      for {
+        sRandom    <- ZIO.effectTotal(new SRandom(seed))
+        testRandom <- TestRandom.makeTest(DefaultData)
+        _          <- testRandom.setSeed(seed)
+        actual     <- testRandom.nextGaussian
+        expected   <- ZIO.effectTotal(sRandom.nextGaussian)
+      } yield assert(math.abs(actual - expected), isLessThan(0.01))
```

Comment:
You can use `Assertion.approximatelyEquals` here.

Reply:
much nicer!